### DISCUSSION TITLE
Cannot figure out how to set the precision of comparison on DataTable level for numeric fields

### DISCUSSION THREAD

**Discussion by paulinuk** [fluentassertions/fluentassertions#1759] [Q&A] [answered]
Hello

I have had massive issues with FluentAssertions trying to compare 2 tables.

I can use the syntax below when comparing 2 objects which works perfectly
```cs
actualTrade.Should().BeEquivalentTo(expectedTrade, options => options
    .Excluding(x => x.Number)
    .Using<decimal>(ctx => ctx.Subject.Should().BeApproximately(ctx.Expectation, 0.8M))
    .WhenTypeIs<decimal>());
```

How can I use this convention when comparing 2 tables?

I tried going through all the numeric columns in the tables and rounding them to a set precision but annoyingly I have 1 row which didn't round the same way which then failed the test

Can someone help please?

Paul

**Comment by dennisdoomen** [comment]
Can you be a bit more specific? What is the actual type of the table object you're comparing?

**Reply by paulinuk** [reply]
Sorry

It’s a DataTable

The using approach works well for simple objects

I want to be able to do that at either the data table or data row level

**Reply by dennisdoomen** [reply]
I don't understand your problem. When you compare `DataTable` instances using `BeEquivalentTo`, you can use the options to customize the comparison, exactly as in your first example.

**Reply by paulinuk** [reply]
I tried the logic below and I get the error 

```
Error CS1061: 'IDataEquivalencyAssertionOptions<DataTable>' does not contain a definition for 'Using' and no accessible extension method 'Using' accepting a first argument of type 'IDataEquivalencyAssertionOptions<DataTable>' could be found (are you missing a using directive or an assembly reference?) (2472, 25)
```

```cs
actual.Should().BeEquivalentTo(expected, options =>
    options.Using<decimal>(ctx => ctx.Subject.Should().BeApproximately(ctx.Expectation, 0.8M))
        .WhenTypeIs<decimal>());
```

**Comment by dennisdoomen** [comment, marked as answer]
Ah, you're right. You've run into a missing feature. The `IDataEquivalencyAssertionOptions` doesn't expose that member, even though the underlying implementing type _does_. I haven't used a `DataTable` or `DataSet` in over 15 years, but we would welcome a PR if you really need it.

**Reply by paulinuk** [reply]
Yes I really do need it please

Do you mean for me to try to work out how to do it in the source code?

**Comment by dennisdoomen** [comment]
Sure.

**Reply by paulinuk** [reply]
Ok I wouldn’t know where to start lol

I will see
